packages/meditrak-app/app/entityMenu/EntityList.js

```diff
@@ -119,54 +223,38 @@ export class EntityList extends PureComponent {
             </TouchableOpacity>
           )}
         </View>
-        {!listData || listData.length === 0 ? (
-          <Text style={localStyles.noResultsText}>
-            {listData.length === 0 ? `No entities matching '${searchTerm}'.` : 'Loading...'}
-          </Text>
-        ) : (
-          <FlatList
-            data={listData}
-            renderItem={this.renderEntityCell}
-            keyExtractor={item => item.id}
-            keyboardShouldPersistTaps="always"
-            initialNumToRender={50}
-            maxToRenderPerBatch={50}
-            windowSize={50}
-            // This allows fast scrolling of the entire list, without getting the layout the
-            // list (especially on slower devices) will stop scrolling before the end is reached
-            // and the device will take a moment to increase the scroll height before continuing.
-            getItemLayout={(data, index) => ({
-              length: ITEM_HEIGHT,
-              offset: ITEM_HEIGHT * index,
-              index,
-            })}
-            ref={listComponent => {
-              this.listComponent = listComponent;
-            }}
-            style={localStyles.container}
-          />
-        )}
+        {this.renderResults()}
       </View>
     );
   }
 }
 
 EntityList.propTypes = {
-  filteredEntities: PropTypes.array.isRequired,
+  filteredEntities: PropTypes.array,
   selectedEntityId: PropTypes.string,
   onRowPress: PropTypes.func.isRequired,
+  onClear: PropTypes.func.isRequired,
   onMount: PropTypes.func,
+  isOnlyQuestionOnScreen: PropTypes.bool.isRequired,
```

Comment:
I know it's a bit more code, but could we maybe move the isOnlyQuestionOnScreen prop up the stack by one to `EntityQuestion`? Then we could have it pass it in as a `handlesScrollControl` (or something better named) prop? It's cosmetic, but I feel the naming of this prop doesn't make the most sense in this component's context?

Reply:
I'm totally on board with that!

Reply:
Really happy with how that came out, also led me to make the handling of scroll control within the component a bit more contained and obvious